### DISCUSSION TITLE
`Partitioned` cookie attribute

### DISCUSSION THREAD

**Discussion by Flecart** [encode/starlette#2500] [Potential Issue] [answered]
Currently there is experimentation for a `Partitioned` attribute for the cookies. [See docs here](https://developer.mozilla.org/en-US/docs/Web/Privacy/Partitioned_cookies).

This attribute is important for third party cookies to work when in the future `SameSite=None; Secure` cookies will stop working
For example Chrome is starting to block them for some users https://developers.google.com/privacy-sandbox/3pcd#report-issues.
Example: this is a common warning for third party cookies on Chrome:
```txt
Cookies with the SameSite=None; Secure and not Partitioned attributes that operate in cross-site contexts are third-party cookies. In future Chrome versions, reading third-party cookies will be blocked. This behavior protects user data from cross-site tracking.
Please refer to the article linked to learn more about preparing your site to avoid potential breakage.
```

So I would like to suggest to add this attribute for the cookie to prepare for the future blocks on standard third party cookies.

Useful links:
https://datatracker.ietf.org/doc/html/draft-cutler-httpbis-partitioned-cookies#section-2.1
https://github.com/privacycg/CHIPS
https://github.com/python/cpython/issues/112713

**Comment by Kludex** [comment]
This would need a PR to just add the attribute?

**Reply by Flecart** [reply]
I delved a little deeper on the technical implementation.
From starlette point of view yes, it's just adding that attribute.

But I've checked just now that the attribute is still not present on the default library `http`. See [Here](https://github.com/python/cpython/blob/main/Lib/http/cookies.py#L275).
So we first need the change on that lib, the PR is open there: https://github.com/python/cpython/pull/112714 but it isn't merged yet.

I opened the draft PR on starlette in the meantime, see #2501.

**Reply by Kludex** [reply]
Watching there. Thanks! 👍

**Comment by Kludex** [comment, marked as answer]
I've created https://github.com/encode/starlette/issues/2507 from this discussion.

**Comment by Flecart** [comment]
Reopening for https://github.com/encode/starlette/pull/2501
